src/components/Filters.js

```diff
@@ -0,0 +1,64 @@
+
+const Filters = props => {
+
+    const displayFruit = () => {
+        const fruits = props.storeItems.filter(x => x.type === 'Fruit')
+        props.setStoreItems(fruits)
+    }
+    const displayVegetables = () => {
+        const vegetables = props.storeItems.filter(x => x.type === 'Vegetable')
+        props.setStoreItems(vegetables)
+    }
+    const resetFilter = () => {
+        props.setStoreItems(props.initialStoreItems)
+    }
+
+    const alphabeticalAToZ = sort => {
+        const sorted = props.storeItems.sort((a, b) => {
+            if (a.name.toLocaleLowerCase() < b.name.toLocaleLowerCase()) return -1
+            if (a.name.toLocaleLowerCase() > b.name.toLocaleLowerCase()) return 1
+            return 0
+        })
+        return sorted
+    }
+
+    const alphabeticalZToA = sort => {
```

Comment:
issue here might be that .sort array method isn't a 'pure' function like map, reduce, filter - it actually modifies the original array. So the 'sorted' you're passing to setStoreItems is the same array, and I think this messes with React functionality.

I got it working by calling .sort at the rendering stage. Not 100% sure this is best practice though.

Reply:
yeah i was also questioning this perhaps its something to do with the the array i get from sort, might have to change my approach on this part. thanks Paul!